dask/dataframe/io/csv.py

```diff
@@ -137,7 +139,7 @@ def __call__(self, part):
             path_info,
         )
         if project_after_read:
-            return df[self.columns]
+            return df[self._columns]
```

Comment:
I'm a little confused about all these `columns` changes.  Is the goal to have `self.columns` always be a list?

Also I think this should be `self.columns` since `self._columns` can theoretically be None right?

Reply:
>I'm a little confused about all these columns changes. Is the goal to have self.columns always be a list?

These changes just make it easier to conform to the new `DataFrameIOFunction` `Protocol` for  column-projection. The subtle differences between `self.columns` and `self.full_columns` for csv made this tricky before.

>Also I think this should be self.columns since self._columns can theoretically be None right?

Yep - Good call!